src/modules/endpoint/__tests__/DatabaseConnector.spec.js

```diff
@@ -292,4 +300,172 @@ describe('DatabaseConnector', () => {
+        muenchen: {
+          languages: {}, last_usage: '2010-05-04T00:00:00.000Z'
+        },
+        dortmund: {
+          languages: {}, last_usage: '2011-05-04T00:00:00.000Z'
+        },
+        ansbach: {
+          languages: {}, last_usage: '2012-05-04T00:00:00.000Z'
+        },
+        augsburg: {
+          languages: {}, last_usage: '2014-05-04T00:00:00.000Z'
+        },
+        regensburg: {
+          languages: {}, last_usage: '2013-05-04T00:00:00.000Z'
+        }
+      }), '')
+
+      await databaseConnector.deleteOldFiles(new DatabaseContext('augsburg'))
+
+      await expectCityFilesExist('muenchen', false)
+      await expectCityFilesExist('dortmund', false)
+      await expectCityFilesExist('ansbach')
+      await expectCityFilesExist('regensburg')
+      await expectCityFilesExist('augsburg')
+
+      expect(JSON.parse(await RNFetchBlob.fs.readFile(databaseConnector.getMetaCitiesPath(), '')))
+        .toEqual({
+          ansbach: {
+            languages: {}, last_usage: '2012-05-04T00:00:00.000Z'
+          },
+          augsburg: {
+            languages: {}, last_usage: '2014-05-04T00:00:00.000Z'
+          },
+          regensburg: {
+            languages: {}, last_usage: '2013-05-04T00:00:00.000Z'
+          }
+        })
+    })
+
+    it('should not delete the resource cache of the same city', async () => {
+      await populateCityContent('augsburg')
+      await populateCityContent('dortmund')
+      await populateCityContent('ansbach')
+      await populateCityContent('regensburg')
+
+      // We have to write this manually, since this is normally done in storeLastUsage, but it calls deleteOldFiles
+      await RNFetchBlob.fs.writeFile(databaseConnector.getMetaCitiesPath(), JSON.stringify({
+        augsburg: {
+          languages: {}, last_usage: '2010-05-04T00:00:00.000Z'
+        },
+        dortmund: {
+          languages: {}, last_usage: '2011-05-04T00:00:00.000Z'
+        },
+        ansbach: {
+          languages: {}, last_usage: '2012-05-04T00:00:00.000Z'
+        },
+        regensburg: {
+          languages: {}, last_usage: '2013-05-04T00:00:00.000Z'
+        }
+      }), '')
+
+      await databaseConnector.deleteOldFiles(new DatabaseContext('augsburg'))
+
+      await expectCityFilesExist('dortmund', false)
+      await expectCityFilesExist('ansbach')
+      await expectCityFilesExist('regensburg')
+      await expectCityFilesExist('augsburg')
+
+      expect(JSON.parse(await RNFetchBlob.fs.readFile(databaseConnector.getMetaCitiesPath(), '')))
+        .toEqual({
+          ansbach: {
+            languages: {}, last_usage: '2012-05-04T00:00:00.000Z'
+          },
+          augsburg: {
+            languages: {}, last_usage: '2010-05-04T00:00:00.000Z'
+          },
+          regensburg: {
+            languages: {}, last_usage: '2013-05-04T00:00:00.000Z'
+          }
+        })
+    })
+  })
 })
```

Comment:
I think you're missing a test that `storeLastUsage` removes old data

Reply:
I think I addest a test which checks that `deleteOldFiles` is called, which should be sufficient since `deleteOldFIles` is tested separately. Should I rather add a separate test too?

Reply:
As you want. Id prefer a separate test, since spying on functions of the same class is somewhat bad style